compute/cloud-client/src/test/java/compute/disks/DisksFromSourceIT.java

```diff
@@ -0,0 +1,267 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+@Timeout(value = 10, unit = TimeUnit.MINUTES)
+public class DisksFromSourceIT {
+
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static String ZONE;
+  private static String REGION;
+  private static CryptoKey CRYPTO_KEY;
+  private static Image DEBIAN_IMAGE;
+  private static String KMS_KEYRING_NAME;
+  private static String KMS_KEY_NAME;
+  private static String KMS_ENCRYPTED_DISK_NAME;
+  private static String KMS_CLONE_ENCRYPTED_DISK_NAME;
+  private static String DISK_FROM_IMAGE;
+  private static String DISK_FROM_DISK;
+  private static String DISK_NAME_REGIONAL;
+  private static String SNAPSHOT_NAME_REGIONAL;
+  private static String DISK_TYPE;
+
+  private ByteArrayOutputStream stdOut;
+
+  // Check if the required environment variables are set.
+  public static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+        .that(System.getenv(envVarName)).isNotEmpty();
+  }
+
+  @BeforeAll
+  public static void setup()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    final PrintStream out = System.out;
+    ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+    // requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+
+    ZONE = "us-central1-a";
+    REGION = "us-central1";
+    KMS_KEYRING_NAME = "compute-test-keyring";
+    KMS_KEY_NAME = "compute-test-key";
+
+    String uuid = UUID.randomUUID().toString().split("-")[0];
+    KMS_ENCRYPTED_DISK_NAME = "test-disk-name-kms-enc" + uuid;
+    KMS_CLONE_ENCRYPTED_DISK_NAME = "test-disk-name-kms-clone-enc" + uuid;
+    DISK_FROM_IMAGE = "test-disk-from-image" + uuid;
+    DISK_FROM_DISK = "test-disk-from-disk" + uuid;
+    DISK_NAME_REGIONAL = "test-disk-name-regional" + uuid;
+    SNAPSHOT_NAME_REGIONAL = "test-snapshot-name-from-source" + uuid;
+    DISK_TYPE = String.format("zones/%s/diskTypes/pd-standard", ZONE);
+
+    // Cleanup existing stale instances.
+    Util.cleanUpExistingInstances("test-disk", PROJECT_ID, ZONE);
+
+    // Create disk from image.
+    DEBIAN_IMAGE = null;
+    try (ImagesClient imagesClient = ImagesClient.create()) {
+      DEBIAN_IMAGE = imagesClient.getFromFamily("debian-cloud", "debian-11");
+    }
+
+    // Create KMS Encrypted disk.
+    // The service account needs to have the cloudkms.cryptoKeyVersions.useToEncrypt
+    // permission to execute this test.
+    CRYPTO_KEY = createKmsKey();
+    CreateKmsEncryptedDisk.createKmsEncryptedDisk(PROJECT_ID, ZONE, KMS_ENCRYPTED_DISK_NAME,
```

Comment:
Shouldn't this be in its own test? Some examples from java-pubsub: https://github.com/googleapis/java-pubsub/blob/main/samples/snippets/src/test/java/pubsub/AdminIT.java

Reply:
Compute samples run out of quota because of the huge number of resources being created for testing. Hence, I put them within the startup function as they can be reused.
Else, I will have to recreate resources each time or combine all the tests into a single test method (which can be confusing).
wdyt?

Reply:
Is it possible to add an auto teardown during creation? https://cloud.google.com/compute/docs/instances/schedule-instance-start-stop#creating_an_instance_schedule

Reply:
Yes, for sure. But they would significantly increase the execution time of each test case, as we would wait for the creation/ deletion to complete before proceeding to the next testcase.
The time taken for the build to run might increase upto 1.5x.

Reply:
I was going to suggest running the tests in parallel (https://www.baeldung.com/maven-junit-parallel-tests#1-using-parallel-parameter), which can be a solution.
I haven't done so myself because Pub/Sub resource creation takes very little time.